ext/mbstring/mbstring.c

```diff
@@ -1027,9 +1027,18 @@ static void *_php_mb_compile_regex(const char *pattern)
 /* {{{ _php_mb_match_regex */
 static int _php_mb_match_regex(void *opaque, const char *str, size_t str_len)
 {
-	return onig_search((php_mb_regex_t *)opaque, (const OnigUChar *)str,
-			(const OnigUChar*)str + str_len, (const OnigUChar *)str,
-			(const OnigUChar*)str + str_len, NULL, ONIG_OPTION_NONE) >= 0;
+	OnigMatchParam *mp = onig_new_match_param();
```

Comment:
I thought about it, and keeping a per-thread copy of this may be ok, needs to be checked in oniguruma code what it actually is. I think there could be situations though where more than one regexp is being processed at one time (e.g. regex1 calls callback, which invokes regex2) and I am not sure how sharing MatchParam will behave in this case. I'd start with the safe way and if we can optimize later it's ok.